### DISCUSSION TITLE
Questions about the output files in Cardinal turorials

### DISCUSSION THREAD

**Discussion by Ethan-xj** [neams-th-coe/cardinal#865] [General]
Hello everyone! When running the standalone example, I noticed that in addition to turbPipe.f, Cardinal also outputs avgturbPipe.f, rm2turbPipe.f, and rmsturbPipe.f. I would like to know what these additional output files represent and how they are generated through which settings.

**Comment by aprilnovak** [comment]
Hi @Ethan-xj, those are optional output files which NekRS writes which contain various postprocessed turbulent flow fields. If velocity is represented as a time-average plus a fluctuation,

```
u = <u> + u'
```

Then those postprocessed files contain:

- `rms`: `<u^2>`
- `avg`: `<u>`
- `rm2`: `<uv>`, `<uw>`, etc.

These files are created by adding the `plugins/tavg.hpp` in the `.udf` file and calling the functions with `tavg` (you can see an example for this in the `standalone/turbPipe.udf` file.

**Reply by Ethan-xj** [reply]
Thank you! And I happened to encounter another issue: I cannot download using the script download-openmc-cross-sections.sh.
Here is the error info:
```
(base) ethan@ethan-pc:~/cardinal$ ./scripts/download-openmc-cross-sections.sh
Downloading cross sections to /home/ethan/cross_sections/endfb-vii.1-hdf5
xz: (stdin): File format not recognized
tar: Child returned status 1
tar: Error is not recoverable: exiting now
```
Btw, I am unable to access the openmc-data website.

**Reply by aprilnovak** [reply]
Hmm, that is strange. You cannot access the following?

https://openmc.org/

**Reply by Ethan-xj** [reply]
Thank you and I download the endfb-vii.1-hdf5 manually.
